pkg/bootstrap/static.go

```diff
@@ -0,0 +1,392 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bootstrap
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/auth"
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/cluster"
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/listener"
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/route"
+	bootstrap "github.com/envoyproxy/go-control-plane/envoy/config/bootstrap/v2"
+	accesslog "github.com/envoyproxy/go-control-plane/envoy/config/filter/accesslog/v2"
+	hcm "github.com/envoyproxy/go-control-plane/envoy/config/filter/network/http_connection_manager/v2"
+	trace "github.com/envoyproxy/go-control-plane/envoy/config/trace/v2"
+	"github.com/envoyproxy/go-control-plane/pkg/util"
+	"github.com/ghodss/yaml"
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/types"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	mixerv1 "istio.io/api/mixer/v1"
+	mixerclient "istio.io/api/mixer/v1/config/client"
+)
+
+// Default constants
+const (
+	ZipkinCluster    = "zipkin"
+	TelemetryCluster = "mixer_report_server"
+)
+
+var (
+	circuitBreakers = &cluster.CircuitBreakers{
+		Thresholds: []*cluster.CircuitBreakers_Thresholds{{
+			Priority:           core.RoutingPriority_DEFAULT,
+			MaxConnections:     &types.UInt32Value{Value: 100000},
+			MaxPendingRequests: &types.UInt32Value{Value: 100000},
+			MaxRequests:        &types.UInt32Value{Value: 100000},
+			MaxRetries:         &types.UInt32Value{Value: 3},
+		}},
+	}
+)
+
+// Options are high-level options for the bootstrap.
+type Options struct {
+	Upstreams                      []Upstream // list of upstreams behind the proxy
+	TelemetryUpstream              *Upstream  // set to one of the upstreams to route telemetry locally
+	TelemetryAddress, TelemetrySAN string     // set to telemetry DNS address and SAN, e.g. istio-telemetry:15004
+	ZipkinAddress                  string     // set to non-empty DNS zipkin address, e.g. zipkin:9411
```

Comment:
Should this be `TracingAddress` given we support Jaeger as well?

Reply:
Yes, but Jaeger is simply not wired throughout. Currently, it's zipkin or nothing. I would like too see more options, but that needs to be done in pilot first.